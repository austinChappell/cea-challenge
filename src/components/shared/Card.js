import React from 'react';
import PropTypes from 'prop-types';

const propTypes = {
  children: PropTypes.element.isRequired,
  handleClick: PropTypes.func,
  returnValue: PropTypes.objectOf(PropTypes.any),
};

const defaultProps = {
  handleClick: () => {},
  returnValue: null,
};

const Card = props => (
  <div
    className="Card"
    onClick={() => props.handleClick(props.returnValue)}
  >
    {props.children}
  </div>
);

Card.propTypes = propTypes;
Card.defaultProps = defaultProps;

export default Card;
