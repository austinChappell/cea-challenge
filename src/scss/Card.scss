.Card {
  box-shadow: 2px 2px 8px $color-black;
  box-sizing: border-box;
  padding: 15px;
  flex-basis: 40%;
  margin: 15px;
}