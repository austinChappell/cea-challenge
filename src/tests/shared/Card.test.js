import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';

import Card from '../../components/shared/Card';

describe('The Card Component', () => {
  it('should render correctly', () => {
    const wrapper = shallow(<Card />);
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });
});
