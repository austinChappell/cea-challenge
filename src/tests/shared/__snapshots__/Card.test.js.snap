// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The Card Component should render correctly 1`] = `
<div
  className="Card"
>
  <div />
</div>
`;
